core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3089,4 +3200,142 @@ public Long getValue() {
+
+    @Override
+    public MasterAuditContext setAllowed(boolean allowed) {
+      mAllowed = allowed;
+      return this;
+    }
+
+    /**
+     * Sets mCommand field.
+     *
+     * @param command the command associated with this {@link alluxio.master.Master}
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setCommand(String command) {
+      mCommand = command;
+      return this;
+    }
+
+    /**
+     * Sets mSrcPath field.
+     *
+     * @param srcPath the source path of the command
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setSrcPath(AlluxioURI srcPath) {
+      mSrcPath = srcPath;
+      return this;
+    }
+
+    /**
+     * Sets mDstPath field.
+     *
+     * @param dstPath the destination path of the command
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setDstPath(AlluxioURI dstPath) {
+      mDstPath = dstPath;
+      return this;
+    }
+
+    /**
+     * Sets mUser field.
+     *
+     * @param user the client user name of the authenticated client user of this thread
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setUser(String user) {
+      mUser = user;
+      return this;
+    }
+
+    /**
+     * Sets mIp field.
+     *
+     * @param ip the IP of the client
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setIp(String ip) {
+      mIp = ip;
+      return this;
+    }
+
+    /**
+     * Sets mSrcInode field.
+     *
+     * @param srcInode the source inode of this operation
+     * @return this {@link AuditContext} instance
+     */
+    public MasterAuditContext setSrcInode(Inode srcInode) {
+      mSrcInode = srcInode;
+      return this;
+    }
+
+    /**
+     * Constructor of {@link MasterAuditContext}.
+     *
+     * @param asyncAuditLogWriter
+     */
+    private MasterAuditContext(AsyncUserAccessAuditLogWriter asyncAuditLogWriter) {
+      mAsyncAuditLogWriter = asyncAuditLogWriter;
+      mAllowed = true;
+    }
+
+    @Override
+    public void close() {
+      if (mAsyncAuditLogWriter == null) {
+        return;
+      }
+      mAsyncAuditLogWriter.append(this);
+    }
+
+    @Override
+    public String toString() {
+      if (mSrcInode != null) {
+        return String.format("allowed=%b\tuser=%s\tip=%s\tcmd=%s\tsrc=%s\tdst=%s\tperm=%s:%s:%d",
+            mAllowed, mUser, mIp, mCommand, mSrcPath, mDstPath, mSrcInode.getOwner(),
+            mSrcInode.getGroup(), mSrcInode.getMode());
+      } else {
+        return String.format("allowed=%b\tuser=%s\tip=%s\tcmd=%s\tsrc=%s\tdst=%s\tperm=null",
+            mAllowed, mUser, mIp, mCommand, mSrcPath, mDstPath);
+      }
+    }
+  }
+
+  /**
+   * Creates a {@link MasterAuditContext} instance.
+   * @param command
+   * @param srcPath
+   * @param dstPath
+   * @param srcInode
+   * @return
+   * @throws AccessControlException
```

Comment:
Because AuthenticatedClientUser#getClientUser() can throw AccessControlException, in which case the user is null. Theoretically the system should not reach this state while processing user requests. If it occurs, we throw AccessControlException to denote that something has gone wrong.